src/Language/PureScript/Names.hs

```diff
@@ -51,12 +51,21 @@ freshIdent' = liftM (GenIdent Nothing) fresh
 -- |
 -- Proper names, i.e. capitalized names for e.g. module names, type//data constructors.
 --
-newtype ProperName = ProperName { runProperName :: String } deriving (Show, Read, Eq, Ord, Data, Typeable)
+newtype ProperName a = ProperName { runProperName :: String }
+  deriving (Show, Read, Eq, Ord, Data, Typeable)
+
+data TypeName = TypeName deriving (Show, Read, Eq, Ord, Data, Typeable)
```

Comment:
Possibly, yeah, but I think it might interfere with `Data`, `Typeable`, and aeson's instances, as even defining these with no constructor (`EmptyDataDecls`) caused problems for them.